das-dashboard/backend/services/config_services.py

```
@@ -1,101 +1,170 @@
+        configuration_entries: ConfigurationEntriesDto | None = None,
+    ) -> dict:
+        nested = self._build_export_config(configuration_entries)
+        known_ips = {target["ip"] for target in self.web_config.map_hosts(nested)}
 
-    async def save_config(self, config_file: UploadFile):
-        result = None
         try:
-            os.makedirs(CONFIG_DIR, exist_ok=True)
-            content = await config_file.read()
+            if ip not in known_ips:
+                raise ValueError()
```

Ruff (0.15.18)

[warning] Abstract `raise` to an inner function

(TRY301)